packages/@react-stately/color/src/useHexColorFieldState.ts

```diff
@@ -0,0 +1,154 @@
+  const [isValid, setIsValid] = useState(initialInputIsValid);
+
+  const increment = () => {
+    setColorInputValue((previousValue) => {
+      const color = typeof previousValue === 'string' ? new Color(previousValue) : previousValue;
+      const colorString = color.toString('hex').substring(1);
+      const colorNumber = parseInt(colorString, 16);
+
+      const maxColor = typeof maxValue === 'string' ? new Color(maxValue) : maxValue;
+      const maxColorString = maxColor.toString('hex').substring(1);
+      const maxColorNumber = parseInt(maxColorString, 16);
+
+      const newValue = `#${Math.min(colorNumber + step, maxColorNumber).toString(16).padStart(6, '0').toUpperCase()}`;
+      const newColor = new Color(newValue);
+      updateValidation(newColor);
+      setInputValue(newValue);
+      return newColor;
+    });
+  };
+
+  const incrementToMax = useCallback(() => {
+    if (maxValue != null) {
+      const maxColor = typeof maxValue === 'string' ? new Color(maxValue) : maxValue;
+      setColorInputValue(maxColor);
+      setInputValue(maxColor.toString('hex'));
+    }
+  }, [maxValue, setColorInputValue, setInputValue]);
+
+  const decrement = () => {
+    setColorInputValue((previousValue) => {
+      const color = typeof previousValue === 'string' ? new Color(previousValue) : previousValue;
+      const colorString = color.toString('hex').substring(1);
+      const colorNumber = parseInt(colorString, 16);
+
+      const minColor = typeof minValue === 'string' ? new Color(minValue) : minValue;
+      const minColorString = minColor.toString('hex').substring(1);
+      const minColorNumber = parseInt(minColorString, 16);
+
+      const newValue = `#${Math.max(colorNumber - step, minColorNumber).toString(16).padStart(6, '0').toUpperCase()}`;
+      const newColor = new Color(newValue);
+      updateValidation(newColor);
+      setInputValue(newValue);
+      return newColor;
+    });
+  };
+
+  const decrementToMin = useCallback(() => {
+    if (minValue != null) {
+      const minColor = typeof minValue === 'string' ? new Color(minValue) : minValue;
+      setColorInputValue(minColor);
+      setInputValue(minColor.toString('hex'));
+    }
+  }, [minValue, setColorInputValue, setInputValue]);
+
+  const setColorValue = (color: Color) => {
+    updateValidation(color);
+    setColorInputValue(color);
+    setInputValue(color.toString('hex'));
+  };
+
+  const updateValidation = (value: Color) => {
+    setIsValid(isInputValueValid(value, maxValue, minValue));
+  };
+
+  const commitInputValue = () => {
+    if (!inputValue.length) { return; }
+    updateValidation(colorValue);
+    setInputValue(colorValue.toString('hex'));
+  };
+
+  return {
+    colorValue,
+    setColorValue,
+    value: null,
+    setValue: null,
+    inputValue,
+    increment,
+    incrementToMax,
+    decrement,
+    decrementToMin,
+    commitInputValue,
+    validationState: !isValid ? 'invalid' : null
```

Comment:
I remembered other form inputs don't necessarily decide their validationState on their own. I wonder if we should do the same here as well.

Reply:
Also, I copied this from NumberField but this kind of suggests the component will not be in a valid state (with the green check icon)

Reply:
Removed `validationState` logic from `useHexColorFieldState`. Will probably remove mentioning it in `useHexColorFieldState` once we get updates from #1029

Reply:
@snowystinger 
don't put validation state logic here

Reply:
Also just add an ignore if it's causing problems, or just leave as is

Reply:
Resolution: Leave as is. Currently not using it in any logic inside the hook